bgl/atom_mapping.py

```diff
@@ -193,25 +191,39 @@ def remove_disconnected_components(mol_a, mol_b, cores):
     """
     filtered_cores = []
     for core in cores:
-        core_a = list(core[:, 0])
-        core_b = list(core[:, 1])
-        g_mol_a = mol_to_mapped_bonds_graph(mol_a, core_a)
-        g_mol_b = mol_to_mapped_bonds_graph(mol_b, core_b)
-
-        largest_cc_a = max(nx.connected_components(g_mol_a), key=len)
-        largest_cc_b = max(nx.connected_components(g_mol_b), key=len)
-
-        # pick the smaller connected mapping
-        new_core_idxs = []
-        if len(largest_cc_a) < len(largest_cc_b):
-            # mol_a has the smaller cc
-            for atom_idx in largest_cc_a:
-                new_core_idxs.append(core_a.index(atom_idx))
-        else:
-            # mol_b has the smaller cc
-            for atom_idx in largest_cc_b:
-                new_core_idxs.append(core_b.index(atom_idx))
-        new_core = core[new_core_idxs]
+
+        new_core = core
```

Comment:
this looks right to me, but random questions

1) does the order removal of the smaller components matter?
2) is the number of disconnected components guaranteed to monotonically decrease each round? (in either mol_a and mol_b)

Reply:
1) We remove all smaller components at the same time (for the chosen mol).
2) It's not this strict. There are cases where you have len(cc_a) == 2, len(cc_b) == 1 in one round then len(cc_a) == 1 ,  len(cc_b) == 2 in the next. Or even len(cc_a) == 4, len(cc_b) == 2 then len(cc_a) == 1, len(cc_b) == 3.

It depends on the details of the underlying mapping. In practice, all tests cases are fixed within 4 iterations.